Graphs/XYGraph.js

```diff
@@ -123,6 +124,12 @@ export default class XYGraph extends AbstractGraph {
             .tickSizeInner(xTickGrid ? -this.getAvailableHeight() : xTickSizeInner)
             .tickSizeOuter(xTickSizeOuter);
 
+        if (xTicksLabel && typeof xTicksLabel === 'object') {
+            this.axis.x
+            .tickValues(Object.keys(xTicksLabel))
+            .tickFormat( value => xTicksLabel[value] || null);
+        }
+            
         if(xTickFormat){
```

Comment:
maybe we need an else here in case someone will do both: xTicksLabel and xTickFormat!!!